solutions/cf-588E.mdx

```diff
@@ -0,0 +1,164 @@
+---
+id: cf-588E
+source: CF
+title: Duff in the Army
+author: Chongtian Ma
+---
+
+## Explanation
+
+Note that for each two nodes $u$ and $v$, the path between them is just $u$ to
+$lca(u,v)$ and $v$ to $lca(u,v)$. Also note the constraint of $a \leq 10$ for
+each query. To answer each query, we only need to store the $10$ people with the
+smallest ids for the path from $u$ to $lca(u,v)$ and $v$ to $lca(u,v)$, and
+combine them. This can be done with binary lifting, which will store the $10$
+smallest ids for each vertex to their $2^k$'th ancestor.
+
+## Implementation
+
+We can answer each query in $\mathcal{O}(a \log n)$ even though the constant
+factor is pretty bad, but the problem limits does give room for it. Note
+that if you do store more than $10$ smallest ids the problem will yield a MLE!
```

Comment:
```suggestion
**Time Complexity**: $\mathcal{O}(a \log n)$
```

Reply:
the total complexity isn’t a log n, just for each query.
also why did you remove my paragraph lol